### PR TITLE
Replaced chalk by kleur

### PR DESCRIPTION
**Pull request by tinchoz49**
## :memo: Description

Working with the browser integration I saw that Chalk is not compatible for the browser but [kleur](https://github.com/lukeed/kleur) it is. 

I think having a small, faster and browser-compatible module for formatting terminal text could be helpful in moleculer and moleculer-browser.

## :checkered_flag: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] **I have added tests that prove my fix is effective or that my feature works**
- [x] **New and existing unit tests pass locally with my changes**
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by icebob** [comment]
@lukeed could you review it, please?

File: src/middlewares/debugging/transit-logger.js [hunk 1]
```diff
@@ -39,8 +39,8 @@ module.exports = function TransitLoggerMiddleware(opts) {
 		fs.writeFile(path.join(targetFolder, filename), data, () => { /* Silent error */ });
 	}
 
-	const coloringSend = opts.colors && opts.colors.send ? opts.colors.send.split(".").reduce((a,b) => a[b], chalk) : s => s;
-	const coloringReceive = opts.colors && opts.colors.receive ? opts.colors.receive.split(".").reduce((a,b) => a[b], chalk) : s => s;
+	const coloringSend = opts.colors && opts.colors.send ? opts.colors.send.split(".").reduce((a,b) => a[b] || a()[b], kleur) : s => s;
```
Comment (lukeed) [inline src/middlewares/debugging/transit-logger.js]: Interesting 😆 I've not tried this usage/format before

File: src/tracing/exporters/console.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ class ConsoleTraceExporter extends BaseTraceExporter {
 			gaugeWidth: 40
 		});
 		if (!this.opts.colors)
-			chalk.level = 0;
+			kleur.enabled = false;
```
Comment (lukeed) [inline src/tracing/exporters/console.js]: Same as above; could be:

```js
kleur.enabled = !!this.opts.colors;
```

**Comment by tinchoz49** [comment]
Hi @lukeed thanks for review this.

> Also, kleur doesn't work in the browser, but it won't throw errors upon init or anything...

What I'm mean with browser compatible was that the colors in the console devtool of brave and chrome I'm seeing colors :smile: . With chalk I couldn't.

You can check it here: https://glitch.com/edit/#!/test-kleur

File: src/tracing/exporters/console.js [hunk 3]
```diff
@@ -28,8 +28,8 @@ class ConsoleTraceExporter extends BaseTraceExporter {
 			width: 100,
 			gaugeWidth: 40
 		});
-		if (!this.opts.colors)
-			kleur.enabled = false;
+
+		kleur.enabled = !!this.opts.colorse;
```
Comment (lukeed) [inline src/tracing/exporters/console.js]: typo :)

**Comment by lukeed** [comment]
Hey, TIL~!

I tried to run kleur in the browser before, but it didn't work in a previous version. I had (wrongly) determined that ANSI codes didn't work in browsers. Thanks to you, I just discovered that the current version _does_ work 🕺 🎉 

```js
console.log('\x1b[31mHello\x1b[39m')
```

File: src/middlewares/debugging/transit-logger.js (hunk 1 above)
Comment (icebob) [reply src/middlewares/debugging/transit-logger.js]: I don't know it works with kleur :)
The original code converts a string value like `"bold.yellow"` to `chalk.bold.yellow`

**Comment by icebob** [comment]
@tinchoz49 Please don't update Jest snapshots. 
The test case [turn off the colors](https://github.com/moleculerjs/moleculer/blob/next/test/unit/tracing/exporters/console.spec.js#L179), so if Travis prints colors it's a bug in this PR.

**Comment by tinchoz49** [comment]
> @tinchoz49 Please don't update Jest snapshots.
> The test case [turn off the colors](https://github.com/moleculerjs/moleculer/blob/next/test/unit/tracing/exporters/console.spec.js#L179), so if Travis prints colors it's a bug in this PR.

ah ok

File: src/metrics/reporters/console.js [hunk 4]
```diff
@@ -35,7 +35,7 @@ class ConsoleReporter extends BaseReporter {
 		});
 
 		if (!this.opts.colors)
-			chalk.level = 0;
+			kleur.enabled = false;
```
[the remark this answers is not in the document]
Comment (lukeed) [reply src/metrics/reporters/console.js]: My bad :D

File: src/middlewares/debugging/transit-logger.js (hunk 1 above)
Comment (lukeed) [reply src/middlewares/debugging/transit-logger.js]: It works (verified) – I just hadn't thought to use it this way

**Comment by icebob** [comment]
Good job @tinchoz49!
@lukeed thank you for your help with reviewing.

Can I merge it? Is it done?

**Comment by tinchoz49** [comment]
Thanks, glad to help.

**Pull request merged**